services/cd-service/pkg/repository/repository.go

```diff
@@ -111,6 +111,23 @@ type repository struct {
 	backOffProvider func() backoff.BackOff
 }
 
+type WebhookResolver interface {
+	Resolve(insecure bool, req *http.Request) (*http.Response, error)
+}
+
+type DefaultWebhookResolver struct{}
+
+func (r DefaultWebhookResolver) Resolve(insecure bool, req *http.Request) (*http.Response, error) {
+	tr := &http.Transport{
+		// we reach argo from within the cluster, so there's no ssl:
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: insecure,
+		},
+	}
+	client := &http.Client{Transport: tr}
+	return client.Do(req)
+}
+
```

Comment:
the revolution reference can go anywhere in the commit message, so I'd rather put it in the body, not the headline